bundles/org.openhab.binding.salus/src/main/java/org/openhab/binding/salus/internal/handler/CloudBridgeHandler.java

```diff
@@ -0,0 +1,269 @@
+ */
+package org.openhab.binding.salus.internal.handler;
+
+import static java.util.Collections.emptySortedSet;
+import static java.util.Objects.requireNonNullElse;
+import static java.util.concurrent.TimeUnit.MICROSECONDS;
+import static java.util.concurrent.TimeUnit.SECONDS;
+import static org.openhab.core.thing.ThingStatus.OFFLINE;
+import static org.openhab.core.thing.ThingStatus.ONLINE;
+import static org.openhab.core.thing.ThingStatusDetail.COMMUNICATION_ERROR;
+import static org.openhab.core.thing.ThingStatusDetail.CONFIGURATION_ERROR;
+import static org.openhab.core.types.RefreshType.REFRESH;
+
+import java.time.Duration;
+import java.util.Optional;
+import java.util.SortedSet;
+import java.util.concurrent.ScheduledFuture;
+
+import org.eclipse.jdt.annotation.NonNull;
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.openhab.binding.salus.internal.rest.Device;
+import org.openhab.binding.salus.internal.rest.DeviceProperty;
+import org.openhab.binding.salus.internal.rest.GsonMapper;
+import org.openhab.binding.salus.internal.rest.JettyHttpClient;
+import org.openhab.binding.salus.internal.rest.SalusApi;
+import org.openhab.core.common.ThreadPoolManager;
+import org.openhab.core.io.net.http.HttpClientFactory;
+import org.openhab.core.thing.Bridge;
+import org.openhab.core.thing.ChannelUID;
+import org.openhab.core.thing.binding.BaseBridgeHandler;
+import org.openhab.core.types.Command;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.github.benmanes.caffeine.cache.Caffeine;
+import com.github.benmanes.caffeine.cache.LoadingCache;
+
+/**
+ * @author Martin Grześlowski - Initial contribution
+ */
+@NonNullByDefault
+public final class CloudBridgeHandler extends BaseBridgeHandler implements CloudApi {
+    private Logger logger = LoggerFactory.getLogger(CloudBridgeHandler.class.getName());
+    private final HttpClientFactory httpClientFactory;
+    @NonNullByDefault({})
+    private LoadingCache<String, SortedSet<DeviceProperty<?>>> devicePropertiesCache;
+    @Nullable
+    private SalusApi salusApi;
+    @Nullable
+    private ScheduledFuture<?> scheduledFuture;
+
+    public CloudBridgeHandler(Bridge bridge, HttpClientFactory httpClientFactory) {
+        super(bridge);
+        this.httpClientFactory = httpClientFactory;
+    }
+
+    @Override
+    public void initialize() {
+        try {
+            internalInitialize();
+        } catch (Exception ex) {
+            updateStatus(OFFLINE, CONFIGURATION_ERROR, "Cannot start server! " + ex.getMessage());
+        }
+    }
+
+    private void internalInitialize() {
+        var config = this.getConfigAs(CloudBridgeConfig.class);
+        if (!config.isValid()) {
+            updateStatus(OFFLINE, CONFIGURATION_ERROR, "Username or password is not valid");
+            return;
+        }
+        var httpClient = new JettyHttpClient(httpClientFactory.getCommonHttpClient());
```

Comment:
Why? `var` is a part of java syntax

Reply:
There are a milion ways of coding. In this repository it is common to use explicit types and i prefer them too.

Reply:
Srsly? What's next? Switch statements?

Check this talk - really nice comment about `var` https://youtu.be/DHwNR7h3k5Y?si=mBT7rm5hplJ71gWu&t=645

Reply:
There is no rule against using `var` and we have many usages within this repository already. IMHO it should be preferred only when the type is obvious from the RHS (like in the case above).

Reply:
My search in ~400 addons resulted in about 1000 hits. With hundreds of thousand of vars in this repo I concluded this is not common. This PR has ~250 additions so it would add 25%.

There is no rule to forbid the use of ‘var’ and I won’t prevent it, but I do see advantages when seeing the code the first time (when reviewing for instance or creating a PR to unfamiliar code).

Reply:
> My search in ~400 addons resulted in about 1000 hits. With hundreds of thousand of vars in this repo I concluded this is not common. This PR has ~250 additions so it would add 25%.

It makes sense. Open HAB is developed for >10 years and `var` is only available since Java 10 (I'm not sure but probably this version was not used in OH, the closest LTS is 11). People did not have opportunity to use vars